From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 10 Jul 2020 23:31:26 +0200
Subject: wayland: Look up cursor renderer for device on MetaWaylandPointer

Make it explicitly look up the pointer device, instead of implicitly
relying on it.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1403>
---
 src/wayland/meta-wayland-pointer.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/wayland/meta-wayland-pointer.c b/src/wayland/meta-wayland-pointer.c
index eb46ecc..c71a5ee 100644
--- a/src/wayland/meta-wayland-pointer.c
+++ b/src/wayland/meta-wayland-pointer.c
@@ -1168,8 +1168,13 @@ pointer_set_cursor (struct wl_client *client,
 
   if (surface)
     {
+      ClutterBackend *clutter_backend = clutter_get_default_backend ();
+      ClutterSeat *clutter_seat =
+        clutter_backend_get_default_seat (clutter_backend);
+      ClutterInputDevice *device = clutter_seat_get_pointer (clutter_seat);
       MetaCursorRenderer *cursor_renderer =
-        meta_backend_get_cursor_renderer (meta_get_backend ());
+        meta_backend_get_cursor_renderer_for_device (meta_get_backend (),
+                                                     device);
       MetaWaylandCursorSurface *cursor_surface;
 
       cursor_surface = META_WAYLAND_CURSOR_SURFACE (surface->role);
